// crem-lib/esm5/lib/dropdown/dropdown.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownComponent } from './dropdown.component';
import { DxSelectBoxModule } from 'devextreme-angular/ui/select-box';
import { DxTextBoxModule } from 'devextreme-angular/ui/text-box';
import { DxTemplateModule } from 'devextreme-angular/core';
import * as i0 from "@angular/core";
var DropdownModule = /** @class */ (function () {
    function DropdownModule() {
    }
    /** @nocollapse */ DropdownModule.ɵmod = i0.ɵɵdefineNgModule({ type: DropdownModule });
    /** @nocollapse */ DropdownModule.ɵinj = i0.ɵɵdefineInjector({ factory: function DropdownModule_Factory(t) { return new (t || DropdownModule)(); }, imports: [[
                CommonModule,
                DxSelectBoxModule,
                DxTextBoxModule,
                DxTemplateModule
            ]] });
    return DropdownModule;
}());
export { DropdownModule };
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(DropdownModule, { declarations: [DropdownComponent], imports: [CommonModule,
        DxSelectBoxModule,
        DxTextBoxModule,
        DxTemplateModule], exports: [DropdownComponent] }); })();
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(DropdownModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    DxSelectBoxModule,
                    DxTextBoxModule,
                    DxTemplateModule
                ],
                declarations: [DropdownComponent],
                exports: [DropdownComponent],
                schemas: []
            }]
    }], null, null); })();
