// @storybook/components/dist/controls/options/Radio.js
"use strict";

require("core-js/modules/es.array.concat");

require("core-js/modules/es.array.map");

require("core-js/modules/es.function.name");

require("core-js/modules/es.object.keys");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadioControl = void 0;

var _react = _interopRequireDefault(require("react"));

var _theming = require("@storybook/theming");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var RadiosWrapper = _theming.styled.div(function (_ref) {
  var isInline = _ref.isInline;
  return isInline ? {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
    '> * + *': {
      marginLeft: 10
    }
  } : {};
});

var RadioLabel = _theming.styled.label({
  padding: '3px 0 3px 5px',
  lineHeight: '18px',
  display: 'inline-block'
});

var RadioControl = function RadioControl(_ref2) {
  var name = _ref2.name,
      options = _ref2.options,
      value = _ref2.value,
      _onChange = _ref2.onChange,
      isInline = _ref2.isInline;
  return /*#__PURE__*/_react["default"].createElement(RadiosWrapper, {
    isInline: isInline
  }, Object.keys(options).map(function (key) {
    var id = "".concat(name, "-").concat(key);
    var optionValue = options[key];
    return /*#__PURE__*/_react["default"].createElement("div", {
      key: id
    }, /*#__PURE__*/_react["default"].createElement("input", {
      type: "radio",
      id: id,
      name: name,
      value: optionValue || undefined,
      onChange: function onChange(e) {
        return _onChange(name, e.target.value);
      },
      checked: optionValue === value
    }), /*#__PURE__*/_react["default"].createElement(RadioLabel, {
      htmlFor: id
    }, key));
  }));
};

exports.RadioControl = RadioControl;
RadioControl.displayName = "RadioControl";
