// @costar/crem-lib/esm2015/lib/data-grid/data-grid.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/data-grid/data-grid.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import 'devextreme/data/odata/store';
import DataSource from 'devextreme/data/data_source';
export class DataGridService {
    /**
     * @return {?}
     */
    getDataSource() {
        return new DataSource({
            store: {
                type: 'odata',
                url: 'https://js.devexpress.com/Demos/SalesViewer/odata/DaySaleDtoes',
                beforeSend: (/**
                 * @param {?} request
                 * @return {?}
                 */
                (request) => {
                    request.params.startDate = '2018-05-10';
                    request.params.endDate = '2018-05-15';
                })
            }
        });
    }
}
DataGridService.decorators = [
    { type: Injectable }
];
