// crem-lib/esm2015/src/lib/hero-metric/hero-metric.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeroMetricComponent } from './hero-metric.component';
import { MatCardModule } from '@angular/material/card';
import { MatMenuModule } from '@angular/material/menu';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import * as i0 from "@angular/core";
// tslint:disable-next-line: eofline
export class HeroMetricModule {
}
/** @nocollapse */ HeroMetricModule.ɵmod = i0.ɵɵdefineNgModule({ type: HeroMetricModule });
/** @nocollapse */ HeroMetricModule.ɵinj = i0.ɵɵdefineInjector({ factory: function HeroMetricModule_Factory(t) { return new (t || HeroMetricModule)(); }, imports: [[CommonModule, BrowserAnimationsModule, MatCardModule, MatMenuModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(HeroMetricModule, { declarations: [HeroMetricComponent], imports: [CommonModule, BrowserAnimationsModule, MatCardModule, MatMenuModule], exports: [HeroMetricComponent] }); })();
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(HeroMetricModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, BrowserAnimationsModule, MatCardModule, MatMenuModule],
                declarations: [HeroMetricComponent],
                exports: [HeroMetricComponent]
            }]
    }], null, null); })();
