// @storybook/components/dist/blocks/Preview.stories.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

require("core-js/modules/es.object.assign");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withCenteredMulti = exports.withCenteredSingle = exports.withFullscreenMulti = exports.withFullscreenSingle = exports.withToolbarMulti = exports.wide = exports.withToolbar = exports.gridWith3Columns = exports.column = exports.row = exports.single = exports.codeError = exports.codeExpanded = exports.codeCollapsed = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _theming = require("@storybook/theming");

var _Spaced = require("../spaced/Spaced");

var _Preview = require("./Preview");

var _Story = require("./Story");

var _Button = require("../Button/Button");

var sourceStories = _interopRequireWildcard(require("./Source.stories"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Docs/Preview',
  component: _Preview.Preview
};
exports["default"] = _default;

var _ref = /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1");

var codeCollapsed = function codeCollapsed() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    isExpanded: false,
    withSource: sourceStories.jsx().props
  }, _ref);
};

exports.codeCollapsed = codeCollapsed;
codeCollapsed.displayName = "codeCollapsed";

var _ref2 = /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1");

var codeExpanded = function codeExpanded() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    isExpanded: true,
    withSource: sourceStories.jsx().props
  }, _ref2);
};

exports.codeExpanded = codeExpanded;
codeExpanded.displayName = "codeExpanded";

var _ref3 = /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1");

var codeError = function codeError() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    isExpanded: true,
    withSource: sourceStories.sourceUnavailable().props
  }, _ref3);
};

exports.codeError = codeError;
codeError.displayName = "codeError";

var _ref4 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1"));

var single = function single() {
  return _ref4;
};

exports.single = single;
single.displayName = "single";

var _ref5 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 2"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 3"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 4"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 5"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 6"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 7"));

var row = function row() {
  return _ref5;
};

exports.row = row;
row.displayName = "row";

var _ref6 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
  isColumn: true
}, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 2"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 3"));

var column = function column() {
  return _ref6;
};

exports.column = column;
column.displayName = "column";

var _ref7 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
  columns: 3
}, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 1"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 2"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 3"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 4"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 5"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 6"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 7 long long long long long title"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 8"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 9"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 10"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 11"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 12"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 13"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 14"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 15"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 16"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 17"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 18"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 19"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Button 20"));

var gridWith3Columns = function gridWith3Columns() {
  return _ref7;
};

exports.gridWith3Columns = gridWith3Columns;
gridWith3Columns.displayName = "gridWith3Columns";

var _ref8 = /*#__PURE__*/_react["default"].createElement(_Button.Button, {
  secondary: true
}, "Hello Button");

var buttonFn = function buttonFn() {
  return _ref8;
};

buttonFn.displayName = "buttonFn";

var _ref9 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
  withToolbar: true
}, /*#__PURE__*/_react["default"].createElement(_Story.Story, {
  inline: true,
  storyFn: buttonFn,
  title: "with toolbar"
}));

var withToolbar = function withToolbar() {
  return _ref9;
};

exports.withToolbar = withToolbar;
withToolbar.displayName = "withToolbar";
var Horizontal = (0, _theming.styled)(function (props) {
  return /*#__PURE__*/_react["default"].createElement(_Spaced.Spaced, _extends({
    col: 1
  }, props));
})({
  display: 'grid',
  gridTemplateColumns: '100px calc(100vw + 100px) 100px'
});

var _ref10 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
  withToolbar: true
}, /*#__PURE__*/_react["default"].createElement(Horizontal, null, /*#__PURE__*/_react["default"].createElement("div", null, "START"), /*#__PURE__*/_react["default"].createElement("div", null, "middle"), /*#__PURE__*/_react["default"].createElement("div", null, "END")));

var wide = function wide() {
  return _ref10;
};

exports.wide = wide;
wide.displayName = "wide";

var _ref11 = /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
  withToolbar: true
}, /*#__PURE__*/_react["default"].createElement(_Story.Story, {
  inline: true,
  storyFn: buttonFn,
  title: "story1"
}), /*#__PURE__*/_react["default"].createElement(_Story.Story, {
  inline: true,
  storyFn: buttonFn,
  title: "story2"
}));

var withToolbarMulti = function withToolbarMulti() {
  return _ref11;
};

exports.withToolbarMulti = withToolbarMulti;
withToolbarMulti.displayName = "withToolbarMulti";

var withFullscreenSingle = function withFullscreenSingle() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    withToolbar: true
  }, /*#__PURE__*/_react["default"].createElement(_Story.Story, {
    inline: true,
    storyFn: buttonFn,
    title: "story1",
    parameters: {
      layout: 'fullscreen'
    }
  }));
};

exports.withFullscreenSingle = withFullscreenSingle;
withFullscreenSingle.displayName = "withFullscreenSingle";

var withFullscreenMulti = function withFullscreenMulti() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    withToolbar: true
  }, /*#__PURE__*/_react["default"].createElement(_Story.Story, {
    inline: true,
    storyFn: buttonFn,
    title: "story1",
    parameters: {
      layout: 'fullscreen'
    }
  }), /*#__PURE__*/_react["default"].createElement(_Story.Story, {
    inline: true,
    storyFn: buttonFn,
    title: "story2",
    parameters: {
      layout: 'fullscreen'
    }
  }));
};

exports.withFullscreenMulti = withFullscreenMulti;
withFullscreenMulti.displayName = "withFullscreenMulti";

var withCenteredSingle = function withCenteredSingle() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    withToolbar: true
  }, /*#__PURE__*/_react["default"].createElement(_Story.Story, {
    inline: true,
    storyFn: buttonFn,
    title: "story1",
    parameters: {
      layout: 'centered'
    }
  }));
};

exports.withCenteredSingle = withCenteredSingle;
withCenteredSingle.displayName = "withCenteredSingle";

var withCenteredMulti = function withCenteredMulti() {
  return /*#__PURE__*/_react["default"].createElement(_Preview.Preview, {
    withToolbar: true
  }, /*#__PURE__*/_react["default"].createElement(_Story.Story, {
    inline: true,
    storyFn: buttonFn,
    title: "story1",
    parameters: {
      layout: 'centered'
    }
  }), /*#__PURE__*/_react["default"].createElement(_Story.Story, {
    inline: true,
    storyFn: buttonFn,
    title: "story2",
    parameters: {
      layout: 'centered'
    }
  }));
};

exports.withCenteredMulti = withCenteredMulti;
withCenteredMulti.displayName = "withCenteredMulti";
