// @storybook/components/dist/controls/Boolean.js
"use strict";

require("core-js/modules/es.function.name");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BooleanControl = void 0;

var _react = _interopRequireDefault(require("react"));

var _theming = require("@storybook/theming");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Input = _theming.styled.input({
  display: 'table-cell',
  boxSizing: 'border-box',
  verticalAlign: 'top',
  height: 21,
  outline: 'none',
  border: '1px solid #ececec',
  fontSize: '12px',
  color: '#555'
});

var format = function format(value) {
  return value ? String(value) : null;
};

var parse = function parse(value) {
  return value === 'true';
};

var BooleanControl = function BooleanControl(_ref) {
  var name = _ref.name,
      value = _ref.value,
      _onChange = _ref.onChange;
  return /*#__PURE__*/_react["default"].createElement(Input, {
    id: name,
    name: name,
    type: "checkbox",
    onChange: function onChange(e) {
      return _onChange(name, e.target.checked);
    },
    checked: value
  });
};

exports.BooleanControl = BooleanControl;
BooleanControl.displayName = "BooleanControl";
