// @storybook/components/dist/controls/Number.js
"use strict";

require("core-js/modules/es.function.name");

require("core-js/modules/es.number.constructor");

require("core-js/modules/es.number.is-nan");

require("core-js/modules/es.parse-float");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NumberControl = exports.format = exports.parse = void 0;

var _react = _interopRequireDefault(require("react"));

var _form = require("../form");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var parse = function parse(value) {
  var result = parseFloat(value);
  return Number.isNaN(result) ? null : result;
};

exports.parse = parse;

var format = function format(value) {
  return value != null ? String(value) : '';
};

exports.format = format;

var NumberControl = function NumberControl(_ref) {
  var name = _ref.name,
      value = _ref.value,
      onChange = _ref.onChange,
      min = _ref.min,
      max = _ref.max,
      step = _ref.step;

  var handleChange = function handleChange(event) {
    onChange(name, parse(event.target.value));
  };

  return /*#__PURE__*/_react["default"].createElement(_form.Form.Input, {
    value: value,
    type: "number",
    name: name,
    min: min,
    max: max,
    step: step,
    onChange: handleChange,
    size: "flex"
  });
};

exports.NumberControl = NumberControl;
NumberControl.displayName = "NumberControl";
