// @storybook/components/dist/controls/Range.js
"use strict";

require("core-js/modules/es.array.concat");

require("core-js/modules/es.function.name");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RangeControl = void 0;

var _react = _interopRequireDefault(require("react"));

var _theming = require("@storybook/theming");

var _Number = require("./Number");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var RangeInput = _theming.styled.input({
  boxSizing: 'border-box',
  height: 25,
  outline: 'none',
  border: '1px solid #f7f4f4',
  borderRadius: 2,
  fontSize: 11,
  padding: 5,
  color: '#444'
}, {
  display: 'table-cell',
  flexGrow: 1
});

var RangeLabel = _theming.styled.span({
  paddingLeft: 5,
  paddingRight: 5,
  fontSize: 12,
  whiteSpace: 'nowrap'
});

var RangeWrapper = _theming.styled.div({
  display: 'flex',
  alignItems: 'center',
  width: '100%'
});

var RangeControl = function RangeControl(_ref) {
  var name = _ref.name,
      _ref$value = _ref.value,
      value = _ref$value === void 0 ? 50 : _ref$value,
      onChange = _ref.onChange,
      _ref$min = _ref.min,
      min = _ref$min === void 0 ? 0 : _ref$min,
      _ref$max = _ref.max,
      max = _ref$max === void 0 ? 100 : _ref$max,
      _ref$step = _ref.step,
      step = _ref$step === void 0 ? 1 : _ref$step;

  var handleChange = function handleChange(event) {
    onChange(name, (0, _Number.parse)(event.target.value));
  };

  return /*#__PURE__*/_react["default"].createElement(RangeWrapper, null, /*#__PURE__*/_react["default"].createElement(RangeLabel, null, min), /*#__PURE__*/_react["default"].createElement(RangeInput, {
    value: value,
    type: "range",
    name: name,
    min: min,
    max: max,
    step: step,
    onChange: handleChange
  }), /*#__PURE__*/_react["default"].createElement(RangeLabel, null, "".concat(value, " / ").concat(max)));
};

exports.RangeControl = RangeControl;
RangeControl.displayName = "RangeControl";
