// crem-lib/esm5/src/lib/dxedatagrid/dxedatagrid.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DxedatagridComponent } from './dxedatagrid.component';
import { DxedatagridService } from './dxedatagrid.service';
import { DxDataGridModule } from 'devextreme-angular/ui/data-grid';
import { DxBulletModule } from 'devextreme-angular/ui/bullet';
import { DxTemplateModule } from 'devextreme-angular/core';
import * as i0 from "@angular/core";
var DatagridModule = /** @class */ (function () {
    function DatagridModule() {
    }
    /** @nocollapse */ DatagridModule.ɵmod = i0.ɵɵdefineNgModule({ type: DatagridModule });
    /** @nocollapse */ DatagridModule.ɵinj = i0.ɵɵdefineInjector({ factory: function DatagridModule_Factory(t) { return new (t || DatagridModule)(); }, providers: [DxedatagridService], imports: [[CommonModule, DxDataGridModule, DxBulletModule, DxTemplateModule]] });
    return DatagridModule;
}());
export { DatagridModule };
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(DatagridModule, { declarations: [DxedatagridComponent], imports: [CommonModule, DxDataGridModule, DxBulletModule, DxTemplateModule], exports: [DxedatagridComponent] }); })();
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(DatagridModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, DxDataGridModule, DxBulletModule, DxTemplateModule],
                declarations: [DxedatagridComponent],
                exports: [DxedatagridComponent],
                providers: [DxedatagridService]
            }]
    }], null, null); })();
