// @costar/crem-lib/esm5/lib/data-grid/data-grid.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/data-grid/data-grid.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import 'devextreme/data/odata/store';
import DataSource from 'devextreme/data/data_source';
var DataGridService = /** @class */ (function () {
    function DataGridService() {
    }
    /**
     * @return {?}
     */
    DataGridService.prototype.getDataSource = /**
     * @return {?}
     */
    function () {
        return new DataSource({
            store: {
                type: 'odata',
                url: 'https://js.devexpress.com/Demos/SalesViewer/odata/DaySaleDtoes',
                beforeSend: (/**
                 * @param {?} request
                 * @return {?}
                 */
                function (request) {
                    request.params.startDate = '2018-05-10';
                    request.params.endDate = '2018-05-15';
                })
            }
        });
    };
    DataGridService.decorators = [
        { type: Injectable }
    ];
    return DataGridService;
}());
export { DataGridService };
