// crem-lib/esm2015/src/lib/dxebutton/dxebutton.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DxebuttonComponent } from './dxebutton.component';
import { DxButtonModule } from 'devextreme-angular/ui/button';
import * as i0 from "@angular/core";
export class ButtonModule {
}
/** @nocollapse */ ButtonModule.ɵmod = i0.ɵɵdefineNgModule({ type: ButtonModule });
/** @nocollapse */ ButtonModule.ɵinj = i0.ɵɵdefineInjector({ factory: function ButtonModule_Factory(t) { return new (t || ButtonModule)(); }, imports: [[CommonModule, DxButtonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(ButtonModule, { declarations: [DxebuttonComponent], imports: [CommonModule, DxButtonModule] }); })();
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(ButtonModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, DxButtonModule],
                declarations: [DxebuttonComponent]
            }]
    }], null, null); })();
