// @storybook/components/dist/blocks/ArgsTable/ArgsTable.stories.js
"use strict";

require("core-js/modules/es.object.assign");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Empty = exports.Error = exports.SectionsCompact = exports.Sections = exports.Compact = exports.Normal = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ArgsTable = require("./ArgsTable");

var _ArgRow = require("./ArgRow.stories");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = {
  component: _ArgsTable.ArgsTable,
  title: 'Docs/ArgsTable'
};
exports["default"] = _default;
var propsSection = {
  category: 'props '
};
var eventsSection = {
  category: 'events '
};

var Normal = function Normal() {
  return /*#__PURE__*/_react["default"].createElement(_ArgsTable.ArgsTable, {
    rows: {
      stringType: _ArgRow.stringType,
      numberType: _ArgRow.numberType
    }
  });
};

exports.Normal = Normal;
Normal.displayName = "Normal";

var Compact = function Compact() {
  return /*#__PURE__*/_react["default"].createElement(_ArgsTable.ArgsTable, {
    compact: true,
    rows: {
      stringType: _ArgRow.stringType,
      numberType: _ArgRow.numberType
    }
  });
};

exports.Compact = Compact;
Compact.displayName = "Compact";
var sectionRows = {
  a: Object.assign(Object.assign({}, _ArgRow.stringType), {}, {
    table: Object.assign(Object.assign({}, _ArgRow.stringType.table), propsSection)
  }),
  b: Object.assign(Object.assign({}, _ArgRow.numberType), {}, {
    table: Object.assign(Object.assign({}, _ArgRow.stringType.table), propsSection)
  }),
  c: Object.assign(Object.assign({}, _ArgRow.stringType), {}, {
    table: Object.assign(Object.assign({}, _ArgRow.stringType.table), eventsSection)
  })
};

var _ref = /*#__PURE__*/_react["default"].createElement(_ArgsTable.ArgsTable, {
  rows: sectionRows
});

var Sections = function Sections() {
  return _ref;
};

exports.Sections = Sections;
Sections.displayName = "Sections";

var _ref2 = /*#__PURE__*/_react["default"].createElement(_ArgsTable.ArgsTable, {
  compact: true,
  rows: sectionRows
});

var SectionsCompact = function SectionsCompact() {
  return _ref2;
};

exports.SectionsCompact = SectionsCompact;
SectionsCompact.displayName = "SectionsCompact";

var Error = function Error() {
  return /*#__PURE__*/_react["default"].createElement(_ArgsTable.ArgsTable, {
    error: _ArgsTable.ArgsTableError.NO_COMPONENT
  });
};

exports.Error = Error;
Error.displayName = "Error";

var Empty = function Empty() {
  return /*#__PURE__*/_react["default"].createElement(_ArgsTable.ArgsTable, {
    rows: {}
  });
};

exports.Empty = Empty;
Empty.displayName = "Empty";
